hack/generator/pkg/astmodel/type_name_set.go

```diff
@@ -59,12 +59,17 @@ func (ts TypeNameSet) Equals(set TypeNameSet) bool {
 
 // SetUnion returns a new set with all of the names in s1 or s2.
 func SetUnion(s1, s2 TypeNameSet) TypeNameSet {
-	var result TypeNameSet
-	for val := range s1 {
-		result = result.Add(val)
+	return make(TypeNameSet).AddAll(s1).AddAll(s2)
+}
+
+func (ts TypeNameSet) AddAll(other TypeNameSet) TypeNameSet {
+	if ts == nil {
+		ts = make(TypeNameSet)
 	}
```

Comment:
I'm not a fan of this idiom of creating the object on the fly like this. 

I've seen this in code written by C++ devs as a fertile source of bugs - consumer thinks there's a reference to an existing object there, a new one gets silently created, hijinks ensues.

Reply:
However it is how the type works already. I won't change it for this PR. Go also seems to value "uninitialized objects should still work properly"…

Reply:
Actually I can drop this from the PR as I didn't end up using the method.